File: C++/B/54.cpp
#include <iostream>
using namespace std;
int main()
{
    int n, m, sum = 0;
    cin >> n >> m;
    char a[n][n], b[m][m];
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            cin >> a[i][j];
        }
    }
    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cin >> b[i][j];
        }
    }

    for (int i = 0; i < n - m + 1; i++)
    {
        for (int j = 0; j < n - m + 1; j++)
        {
            for (int k = i; k < i + m; k++)
            {
                for (int l = j; l < j + m; l++)
                {
                    if (a[k][l] == b[k - i][l - i])
                        sum++;
                }
            }
            if (sum == m * m)
            {
                cout << "Yes\n";
                return 0;
            }
        }
    }
    cout << "No\n";
}

File: ruby/B/54.rb
n, m = gets.split.map(&:to_i)
arr_a, arr_b, sum = [], [], 0
n.times do
  arr_a << gets.chomp
end
m.times do
  arr_b << gets.chomp
end
for i in 0..n - m
  for j in 0..n - m
    for k in i..i + m - 1
      for l in j..j + m - 1
        sum += 1 if arr_a[k][l] == arr_b[k - i][l - j]
      end
    end
    if sum == m * m
      puts "Yes"
      exit
    end
    sum = 0
  end
end
puts "No"
